lib/x11test.pm

```diff
@@ -569,7 +569,8 @@ sub start_firefox {
 
 sub firefox_check_default {
     # Set firefox as default browser if asked
-    assert_screen [qw(firefox_default_browser firefox_trackinfo firefox_readerview_window firefox_clean)], 120;
+    assert_screen 'firefox-indicator', 90;
+    check_screen 'firefox_default_browser';
```

Comment:
Please avoid `check_screen` with default timeout. The subsequent code already uses a multi-tag assert screen. This should not be reverted

Reply:
`assert_screen [qw(firefox_default_browser firefox_trackinfo firefox_readerview_window firefox_clean)], 120;` 
is not reliable. We have too many needles with tag firefox_clean and this needle will be matched before the default browser window pop up on o.s.d.

So I want to improve the stability by `check_screen 'firefox_default_browser';` after firefox is launched (`assert_screen 'firefox-indicator', 90;` will make sure FF is launched)

Reply:
I don't think this is a good idea. Many people - mainly QAM - invested a lot of time to get the firefox tests working properly so we should not accept a revert lightly. Simply stated: Sorry, `check_screen` with default timeout is not acceptable. I do not understand yet how too many needles for firefox_clean make a problem unless a dialog on the SUT would disappear in time. However, looking at https://openqa.suse.de/admin/needles I can confirm that there *are* many needles for firefox_clean where many of them seem to have never matched. I will try to clean up some of them. Maybe this helps already.